### PR TITLE
feat: close previous versions of the report

### LINKED ISSUE #79: Close previous versions of the report for the same release
The issue title could have the generation date to help identifying how old is the information.

### PR DESCRIPTION
**Pull request by jacobdotcosta**
When launching the report previous instances with the same title will be closed. 

Closes #79

### CODE REVIEW

File: src/main/java/dev/snowdrop/release/App.java [hunk 1]
```diff
@@ -182,9 +182,11 @@ public void listCVEs(
         final var cves = cveService.listCVEs(Optional.ofNullable(version));
         reportStatus(cves);
         if (release) {
+            final String repoName = "snowdrop/reports";
```
Comment (metacosm) [inline src/main/java/dev/snowdrop/release/App.java:185]: Shouldn't this use the same repository (and probably be a constant) from the CveReportingService?

File: src/main/java/dev/snowdrop/release/services/GitService.java [hunk 2]
```diff
@@ -139,4 +141,39 @@ public void createGithubIssue(final String mdText, final String issueTitle, fina
...
+     * @param token
+     * @param repoName
+     * @param version
+     * @throws IOException
+     */
+    public void closeOldCveIssues(final String label, final String token, final String repoName, final String version) throws IOException {
+        final String issueTitle = this.getCveIssueTitle(version);
+        GitHub github = new GitHubBuilder().withOAuthToken(token).build();
+        GHRepository gitHubRepo = github.getRepository(repoName);
+        List<GHIssue> ghIssues = gitHubRepo.getIssues(GHIssueState.OPEN);
```
Comment (metacosm) [inline src/main/java/dev/snowdrop/release/services/GitService.java:157]: There's no other way than to retrieve all the open issues and then filter them out?

**Review by metacosm — CHANGES_REQUESTED** [review]

File: src/main/java/dev/snowdrop/release/services/GitService.java (hunk 2 above)
Comment (jacobdotcosta) [reply src/main/java/dev/snowdrop/release/services/GitService.java:157]: Not that I could find out. I can only see 2 methods for querying issues, one with the state and the other with the state and the milestone.

File: src/main/java/dev/snowdrop/release/reporting/CVEReportingService.java [hunk 3]
```diff
@@ -38,28 +38,20 @@
...
-    /**
-     * <p>Owner of the snowdrop weekly development repository owner</p>
-     */
-    public static String WEEK_DEV_REPO_OWNER = "snowdrop";
+    public final static String CVE_REPORT_REPO_NAME = "snowdrop/reports";
 
-    /**
-     * <p>Name of the snowdrop weekly development repository</p>
-     */
-    public static String WEEK_DEV_REPO_NAME = "team";
```
Comment (metacosm) [inline src/main/java/dev/snowdrop/release/reporting/CVEReportingService.java:57]: I was wondering why `team` was used previously…

File: src/main/java/dev/snowdrop/release/App.java [hunk 4]
```diff
@@ -182,11 +182,10 @@ public void listCVEs(
...
         if (release) {
-            final String repoName = "snowdrop/reports";
             if (Optional.ofNullable(token).isPresent()) {
-                git.closeOldCveIssues("cve",token,repoName, version);
+                git.closeOldCveIssues("cve",token, CVEReportingService.CVE_REPORT_REPO_NAME, version);
                 String mdReport = cveReportSvc.buildMdReport(cves,  git.getCveIssueTitle(version));
-                git.createGithubIssue(mdReport, git.getCveIssueTitle(version), "cve", token, repoName);
+                git.createGithubIssue(mdReport, git.getCveIssueTitle(version), "cve", token, CVEReportingService.CVE_REPORT_REPO_NAME);
             } else {
                 LOG.error("Cannot release CVE to GitHub. Github API token is required if --publish is enabled. PLease specify the github token using --token.");
```
Comment (metacosm) [inline src/main/java/dev/snowdrop/release/App.java:190]:
```suggestion
                LOG.error("Cannot release CVE to GitHub. Github API token is required if --publish is enabled. Please specify the github token using --token.");
```

Comment (metacosm) [reply src/main/java/dev/snowdrop/release/App.java:190]: Hmm, that looks weird with the `<span` stuff in there…

Comment (jacobdotcosta) [reply src/main/java/dev/snowdrop/release/App.java]: fixed.

File: src/main/java/dev/snowdrop/release/App.java [hunk 5]
```diff
@@ -183,10 +183,11 @@ public void listCVEs(
         reportStatus(cves);
         if (release) {
             if (Optional.ofNullable(token).isPresent()) {
-                String mdReport = cveReportSvc.buildMdReport(cves, "Version - " + Optional.ofNullable(version).orElse(" ALL OPEN"));
-                git.createGithubIssue(mdReport, "CVE " + (version != null ? "for " + version : "list"), "cve", token, "snowdrop/reports");
+                git.closeOldCveIssues("cve",token, CVEReportingService.CVE_REPORT_REPO_NAME, version);
```
Comment (metacosm) [inline src/main/java/dev/snowdrop/release/App.java:186]: Note that the method names should also be changed:  `Cve` -> `CVE` but that shouldn't be done as part of this PR.

**Review by metacosm — APPROVED** [review]

**Pull request merged**